var/spack/repos/builtin/packages/libxslt/package.py

```diff
@@ -21,19 +21,37 @@ class Libxslt(AutotoolsPackage):
     version('1.1.28', '9667bf6f9310b957254fdcf6596600b7')
     version('1.1.26', 'e61d0364a30146aaa3001296f853b2b9')
 
-    variant('crypto',  default=True,
-            description='Build libexslt with crypto support')
+    variant('crypto', default=True, description='Build libexslt with crypto support')
+    variant('python', default=False, description='Build Python bindings')
 
-    depends_on("libiconv")
-    depends_on("libxml2")
-    depends_on("xz")
-    depends_on("zlib")
-    depends_on("libgcrypt", when="+crypto")
+    depends_on('pkgconfig@0.9.0:', type='build')
+    depends_on('libiconv')
+    depends_on('libxml2')
+    depends_on('libxml2+python', when='+python')
+    depends_on('xz')
+    depends_on('zlib')
+    depends_on('libgcrypt', when='+crypto')
+
+    extends('python+shared', when='+python')
 
     def configure_args(self):
         args = []
-        if '~crypto' in self.spec:
+
+        if '+crypto' in self.spec:
+            args.append('--with-crypto')
+        else:
             args.append('--without-crypto')
```

Comment:
I think you can reduce this to:
```python
args.extend(
    self.with_or_without('crypto') + \
    self.with_or_without('python', activation_value=lambda x: self.spec[x].home)
)
```

Reply:
I find lambdas more succinct but less readable. Also, still needs documentation 😛